nbconvert/exporters/exporter.py

```diff
@@ -308,4 +308,5 @@ def _preprocess(self, nb, resources):
         #to each preprocessor
         for preprocessor in self._preprocessors:
             nbc, resc = preprocessor(nbc, resc)
+            nbformat.validate(nbc, relax_add_props=True)
```

Comment:
I wonder if it's worth catching the validation error and re-raising with a message saying which preprocessor made it invalid?